esp/bindings/http/platform/httptransport.cpp

```diff
@@ -2541,3 +2633,176 @@ void CHttpResponse::handleExceptions(IXslProcessor *xslp, IMultiException *me, c
     if (handleExceptions(xslp, me, serv, meth, errorXslt) && logHandleExceptions)
         PROGLOG("Exception(s) handled");
 }
+
+bool CHttpResponse::shouldCompress(int& compressType)
+{
+    compressType = 0;
+    if (!m_compressionEnabled)
+        return false;
+    if (m_content_length == 0 || m_content.length() == 0)
+        return false;
+
+#ifdef _USE_ZLIB
+    if (hasHeader(HTTP_HEADER_CONTENT_ENCODING) || hasHeader(HTTP_HEADER_TRANSFER_ENCODING)) //Already encoded/compressed
+        return false;
+
+    static const int DEFAULT_MIN_COMPRESS_LENGTH = 1000;
+    int min_len = DEFAULT_MIN_COMPRESS_LENGTH;
+    IPropertyTree* proccfg = nullptr;
+    IEspServer* espserver = queryEspServer();
+    if (espserver)
+        proccfg = espserver->queryProcConfig();
+    if (proccfg)
+        min_len = proccfg->getPropInt("@minCompressLength", DEFAULT_MIN_COMPRESS_LENGTH);
+    if (m_content.length() < min_len)
+        return false;
+
+    IEspContext * context = queryContext();
+    if (!context)
+        return false;
+    CHttpRequest* request = dynamic_cast<CHttpRequest*>(context->queryRequest());
+    if (!request)
+        return false;
+    StringBuffer acceptEncoding;
+    request->getHeader(HTTP_HEADER_ACCEPT_ENCODING, acceptEncoding);
+    if (acceptEncoding.length() == 0)
+        return false;
+    char* aestr = acceptEncoding.toLowerCase().detach();
```

Comment:
Is it necessary to detach and free?  even if j_strok_r writes nulls it may still clean up in StringBuffer ok?